### PR TITLE
Lecture 1

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request adds the slides support (slidev) and lecture 1.

This still has an issue #38, that will be fixed eventually, but does require additional work.

### Testing Strategy

This pull request was tested by Alexandru Radovici

### TODO or Help Wanted

Feedback

### Build

- [x] Ran `npm build`.

### Author

Signed-off-by: Alexandru Radovici <alexandru.radovici@wyliodrin.com>

### CODE REVIEW

File: slides/lectures/01/admin/slides.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+## Lectures
+  - 12 lectures
+  - 1 Q&A lecture for the project
+
+## Labs
+  - 12 labs
+
+## Project
+- You have to build a project
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/admin/slides.md:16]:
```suggestion
- You have to build hardware a device that runs software written in Rust
```

File: slides/lectures/01/admin/slides.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- Will cost around 150 RON
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/admin/slides.md:17]:
```suggestion
- The cost for the hardware components will be around 150 RON
```

File: slides/lectures/01/admin/slides.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+- Presented at PM Fair during the last week of the semester
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/admin/slides.md:18]: What is PM Fair

File: slides/lectures/01/processor/slides.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+- does not require an Operating System
+
+<img src="/processor/pico.jpg" class="m-10 h-30">
+
+
+:: right ::
+
+# Microprocessor (CPU)
+computes
+
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/processor/slides.md:25]: General purpose, for PC & workstations

File: slides/lectures/01/processor/slides.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+- high operating frequency (GHz)
+- limited number of I/O ports
+- runs application and server software
+- usually requires an Operating System
+
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/processor/slides.md:30]: Usual cost: 75$ - 500$
Annual demand: tens of millions

File: slides/lectures/01/processor/slides.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+layout: section
+---
+# What is a microprocessor?
+
+---
+layout: two-cols
+---
+# Microcontroller (MCU)
+controls hardware
+
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/processor/slides.md:11]: Integrated in embedded systems for certain tasks(not general purpose)

File: slides/lectures/01/processor/slides.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+- low operating frequency (MHz)
+- a lot of I/O ports
+- runs simple software
+- controls hardware
+- does not require an Operating System
+
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/processor/slides.md:17]: Usual cost: 0.1$ - 25$
Annual demand: bilions

**Review by DTeodor-Alexaandru — CHANGES_REQUESTED** [review]

File: slides/lectures/01/admin/slides.md (hunk 3 above)
Comment (alexandruradovici) [reply slides/lectures/01/admin/slides.md:18]: https://ocw.cs.pub.ro/courses/pm/pm/lab/regulament - project presentations at the booths, nu?

File: slides/lectures/01/processor/slides.md (hunk 4 above)
Comment (alexandruradovici) [reply slides/lectures/01/processor/slides.md:25]: Debatable here, some devices do have big chips for computing, like intelligent door handles, they for sure use a CPU.

**Review by DTeodor-Alexaandru — APPROVED** [review]

File: slides/lectures/01/admin/slides.md (hunk 3 above)
Comment (DanutAldea) [reply slides/lectures/01/admin/slides.md:18]: The text is too much now, overflowing the screen.

File: slides/lectures/01/rp2040/slides.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+that interconnects the cores with the peripherals
+
+<div align="center">
+<img src="/rp2040/rp2040_bus.png" class="h-100" />
+</div align="center">
+
+
+| Function | Meaning |
+|----------|---------|
+| GPIO | General Purpose Input/Output |
```
Comment (irina-b-dev) [inline slides/lectures/01/rp2040/slides.md:100]: Cannot see text at the bottom. Image is too big.

**Review by ZamfirAndreiPaul — APPROVED** [review]

**Review by irina-nita — APPROVED** [review]

**Review by DanutAldea — CHANGES_REQUESTED** [review]

File: slides/lectures/01/processor/slides.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+layout: two-cols
+---
+# Microcontroller (MCU)
+Integrated in embedded systems for certain tasks
+
+- low operating frequency (MHz)
+- a lot of I/O ports
+- controls hardware
+- does not require an Operating System
+- cost $0.1 - $25
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/processor/slides.md:16]:
```suggestion
- cost: $0.1 - $25
```

File: slides/lectures/01/processor/slides.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+
+:: right ::
+
+# Microprocessor (CPU)
+General purpose, for PC & workstations
+
+- high operating frequency (GHz)
+- limited number of I/O ports
+- usually requires an Operating System
+- cost $75 - $500
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/processor/slides.md:30]:
```suggestion
- cost: $75 - $500
```

File: slides/lectures/01/processor/slides.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+- annual demand tens of millions
```
Comment (DTeodor-Alexaandru) [inline slides/lectures/01/processor/slides.md:31]:
```suggestion
- annual demand: tens of millions
```

**Review by DTeodor-Alexaandru — APPROVED** [review]

**Review by DTeodor-Alexaandru — DISMISSED** [review]

**Review by ZamfirAndreiPaul — APPROVED** [review]

**Review by DTeodor-Alexaandru — APPROVED** [review]

**Review by DanutAldea — APPROVED** [review]

**Pull request merged**